GeeksCoreLibrary/Core/Middlewares/IpAccessMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using GeeksCoreLibrary.Core.Helpers;
+using GeeksCoreLibrary.Modules.Objects.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace GeeksCoreLibrary.Core.Middlewares;
+
+public class IpAccessMiddleware
+{
+    private readonly RequestDelegate next;
+    private readonly ILogger<IpAccessMiddleware> logger;
+
+    public IpAccessMiddleware(ILogger<IpAccessMiddleware> logger, RequestDelegate next)
+    {
+        this.logger = logger;
+        this.next = next;
+    }
+
+    public async Task Invoke(HttpContext context, IObjectsService objectsService)
+    {
+        logger.LogDebug("Invoked IpAccessMiddleware");
+        var userIp = HttpContextHelpers.GetUserIpAddress(context);
+        var blockedIps = await objectsService.FindSystemObjectByDomainNameAsync("blockips");
+        if (!String.IsNullOrEmpty(blockedIps) && blockedIps.Contains(userIp))
```

Comment:
Sorry, I didn't notice this before, but please also improve this check; You now use "Contains" on the entire string. You should split the string on comma and then look into that array if that contains the IP of the user. Otherwise you might get false positives if users have an IP that is very similar to one in the list.